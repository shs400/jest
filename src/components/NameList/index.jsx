import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NameList extends Component {
  static propTypes = {
    names: PropTypes.arrayOf(PropTypes.string),
  }

  static defaultProps = {
    names: [],
  }

  renderList() {
    const { names } = this.props;
    const nameList = names.map(
      (name, index) => (<li className={`list_${index}`} key={`${name}_${index.toString()}`}>{name}</li>),
    );
    return nameList;
  }

  render() {
    return (
      <ul>
        { this.renderList() }
      </ul>
    );
  }
}

export default NameList;
