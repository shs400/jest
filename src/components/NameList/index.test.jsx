import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import NameList from './index';
configure({ adapter: new Adapter() });

describe('NameList', () => {
  let component = null;

  it('renders correctly', () => {
    component = shallow(<NameList names={["로이", "심현수"]} />);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('list text', () => {
    it('has a li', () => {
      expect(component.find('li').exists()).toBe(true);
    });

    it('has a ul', () => {
      expect(component.find('ul').exists()).toBe(true);
    });
  })
});