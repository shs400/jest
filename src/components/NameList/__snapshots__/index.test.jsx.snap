// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameList matches snapshot 1`] = `
<ul>
  <li
    className="list_0"
    key="로이_0"
  >
    로이
  </li>
  <li
    className="list_1"
    key="심현수_1"
  >
    심현수
  </li>
</ul>
`;
